### PR TITLE
Move from gitbook to docusaurus, build docs in Travis CI

### PR DESCRIPTION
**Pull request by danpaul000**
Moving over from https://github.com/solana-labs/solana/pull/10806 to be working from a remote fork rather than upstream branch.

### CODE REVIEW

**Comment by danpaul000** [comment]
@mvines I could use some help figuring out this Travis build setup.  All the pieces are in place and working except for `docs/build-cli-usage.sh` which requires rust, so I am installing rust in `docs/.travis/before_install.sh`.  The deps `libudev-dev` and `libsystemd-dev` are failing to install thereafter with
```
libsystemd-dev : Depends: libsystemd0 (= 237-3ubuntu10.38) but 237-3ubuntu10.39 is to be installed
libudev-dev : Depends: libudev1 (= 237-3ubuntu10.38) but 237-3ubuntu10.39 is to be installed
E: Unable to correct problems, you have held broken packages.
```

Should this be dockerized, or need some manual massaging?

**Comment by mvines** [comment]
Try adding a docker service declaration to the docs build, like this:
https://github.com/solana-labs/solana/blob/fec7a61f59f5d5fd272932e74f28455742f3eeb3/.travis.yml#L94-L95
and then use `ci/docker-run.sh` to run `docs/build-cli-usage.sh` within the standard rust container.  That should avoid needing to install Rust in the Travis job

**Comment by danpaul000** [comment]
All the infrastructure is now in place for Docusaurus to get built and published on master.  I'll handle backporting to 1.2 after this has some time to soak.  

@mvines the only piece I wasn't able to fully test within travis (though verified locally) is the consumption of an env variable containing a private value (VERCEL_TOKEN).  This is set in the travis settings, but as per the platform, is not accessible when running PR jobs from an external repo (my fork).  It's only needed for the post-commit publication to the final URL.

With the value set locally, the publish ran fine: https://edge.docs.solana.com/

File: docs/README.md [hunk 1]
```diff
@@ -1,31 +1,25 @@
...
+
+Solana's Docs are built using [Docusaurus 2](https://v2.docusaurus.io/).
 
-Run any Rust tests in the markdown:
+### Installation
 
-```bash
-$ make test
+```
+$ yarn
```
Comment (garious) [inline docs/README.md:8]: Is this still true? Or npm now?

**Comment by danpaul000** [comment]
> Really happy to see the diff get so much smaller. I'm curious about the .bob files moving from docs/art to docs/src. Is that really needed? And are the .bob files being rendered as part of the build?

Hmm, looks like the bob files are not rendering, will fix that now.

**Comment by danpaul000** [comment]
@garious rebased, fixed your comments, and the docs build is working.  Should be good to go finally, but for some reason the web.js travis build is now failing, though I haven't touched anything in that area.  Checking recent history.

**Comment by danpaul000** [comment]
@t-nelson @jstarry could one of you guys please help me figure out why the web3.js job is failing here?  It's only running in this otherwise docs-only PR because I touched the .travis.yml file, but it looks like you were looking in to some CI flakyiness in this arena earlier this week and I'm just hitting it now.  https://travis-ci.com/github/solana-labs/solana/jobs/360046251

**Comment by CriesofCarrots** [comment]
> @t-nelson @jstarry could one of you guys please help me figure out why the web3.js job is failing here? It's only running in this otherwise docs-only PR because I touched the .travis.yml file, but it looks like you were looking in to some CI flakyiness in this arena earlier this week and I'm just hitting it now. https://travis-ci.com/github/solana-labs/solana/jobs/360046251

@danpaul000 , unfortunately I think you're getting the spillover from rpc changes here: https://github.com/solana-labs/solana/pull/10984
I'm waiting for some things to run, so I can work on updating web3.

**Comment by mvines** [comment]
Ack sorry.   We should relax the Object type checking in web3.js to permit unknown fields.  I think we're starting to do this but there's still some places where we're too strict.  @CriesofCarrots - I can take this over if you'd like

**Comment by CriesofCarrots** [comment]
> Ack sorry. We should relax the Object type checking in web3.js to permit unknown fields. I think we're starting to do this but there's still some places where we're too strict. @CriesofCarrots - I can take this over if you'd like

@mvines I have the dumb solution (adding blockTime) done, but if you have time to implement relaxed Object type checking instead, that would be great!

**Comment by mvines** [comment]
@danpaul000 - rebase on https://github.com/solana-labs/solana/pull/11005 to get unstuck

**Pull request merged**